src/richie/apps/courses/templates/courses/cms/program_list.html

```diff
@@ -10,14 +10,26 @@ <h1 class="subheader__title">{{ current_page.get_title }}</h1>
 
 {% block content %}
 <div class="program-list">
-    {% for page in current_page.get_child_pages %}
-        {% if page.program %}
-            {% include "courses/cms/fragment_program_glimpse.html" with program=page.program %}
+    <div class="program-list__block">
+        {% if current_page.publisher_is_draft or not current_page|is_empty_placeholder:"maincontent" %}
+        <div class="program-list__maincontent program-list__row">
+            {% placeholder "maincontent" %}
+        </div>
         {% endif %}
-    {% empty %}
-        <p class="program-glimpse program-glimpse--empty">
-            {% trans "No associated programs" %}
-        </p>
-    {% endfor %}
+
+        <div class="program-list__programs program-list__row">
+            <section class="program-glimpse-list">
+                {% for page in current_page.get_child_pages %}
+                    {% if page.program %}
+                        {% include "courses/cms/fragment_program_glimpse.html" with program=page.program %}
+                    {% endif %}
+                {% empty %}
+                    <p class="program-glimpse program-glimpse--empty">
```

Comment:
The empty variant may not be correctly defined in your `program-list` it may not have a fine layout, i think there is a "empty" sass mixin somewhere you could use to do that.

Reply:
I checked and it is well formatted. This element was already there and the css taking care of the empty state is already there.